qm-plugins/qm-object-cache/html/class-qm-output-object-cache-ops.php

```diff
@@ -0,0 +1,212 @@
+		return $class;
+	}
+
+	/**
+	 * @param array<string, mixed[]> $menu
+	 * @return array<string, mixed[]>
+	 */
+	public function panel_menu( array $menu ) {
+		if ( isset( $menu['object_cache'] ) ) {
+			$menu['object_cache']['children'][] = $this->menu( array(
+				'id'    => 'qm-object_cache_ops',
+				'href'  => '#qm-object_cache_ops',
+				'title' => __( 'Operations', 'qm-object-cache' ),
+			));
+		}
+
+		return $menu;
+	}
+
+	/**
+	 * Returns in human readable size format
+	 *
+	 * @param $size int|null     Raw size
+	 * @return $size string Human readable size format
+	 */
+	public function process_size( ?int $size ) {
+		return size_format( $size, 2 );
+	}
+
+	/**
+	 * Returns in human readable time format.
+	 *
+	 * @param float|null $time   Raw time
+	 * @return string $time Human readable time format
+	 */
+	public function process_time( ?float $time ) {
+		return number_format_i18n( sprintf( '%0.1f', $time * 1000 ), 1 ) . 'ms';
+	}
+
+	/**
+	 * Returns in human readable result.
+	 *
+	 * @param $result  string Memcached operation result
+	 * @return $result string Human readable memcached operation result
+	 */
+	public function process_result( string $result ) {
+		switch ( trim( $result ) ) {
+			case 'not_in_memcache':
+				$result = __( 'Not in Memcached', 'qm-object-cache' );
+				break;
+			case 'memcache':
+				$result = __( 'Found in Memcached', 'qm-object-cache' );
+				break;
+			case '[mc already]':
+				$result = __( 'Already in Memcached', 'qm-object-cache' );
+				break;
+			case '[lc already]':
+				$result = __( 'Local cache already', 'qm-object-cache' );
+				break;
+		}
+
+		return $result;
+	}
+
+	/**
+	 * Outputs a sortable table column header.
+	 *
+	 * @param string $title Title to use on column header
+	 */
+	public function output_sortable_table_col( string $title ) {
+		echo '<th scope="col" class="qm-sortable-column" role="columnheader">';
+		echo $this->build_sorter( esc_html( $title ) ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		echo '</th>';
+	}
+
+	/**
+	 * Outputs a filterable table column header.
+	 *
+	 * @param string $title Title to use on column header
+	 * @param array $values Groups to filter by
+	 * @param array $args   Additional arguments to be passed in
+	 */
+	public function output_filterable_table_col( string $title, array $values, $args = [] ) {
+		echo '<th scope="col" class="qm-filterable-column">';
+		echo $this->build_filter( sanitize_title( strtolower( $title ) ), $values, esc_html( $title ), $args ); // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+		echo '</th>';
+	}
+
+	/**
+	 * Outputs a toggleable table cell for arrays.
+	 *
+	 * @param array $array Array to be outputted in table cell
+	 */
+	public function maybe_output_toggle_table_cell_for_array( array $array ) {
+		if ( empty( $array ) ) {
+			return;
+		}
+
+		if ( count( $array ) === 1 ) {
+			echo '<td class="qm-nowrap qm-ltr">' . esc_html( $array[0] ) . '</td>';
```

Comment:
Maybe a bit more DRY to call `$this->output_table_cell()` inside this conditional, but not a big deal.